infrastructure/app.py

```diff
@@ -0,0 +1,167 @@
+import os
+
+import aws_cdk as cdk
+from aws_cdk import (
+    CfnOutput,
+    Duration,
+    Environment,
+    RemovalPolicy,
+    Stack,
+    aws_apigatewayv2,
+    aws_apigatewayv2_integrations,
+    aws_certificatemanager,
+    aws_cloudfront,
+    aws_iam,
+    aws_lambda,
+    aws_logs,
+    aws_s3,
+)
+from config import AppConfig
+from constructs import Construct
+
+
+class FederatedCollectionSearchStack(Stack):
+    def __init__(  # type: ignore
+        self,
+        scope: Construct,
+        app_config: AppConfig,
+        **kwargs,
+    ) -> None:
+        id = f"{app_config.project_id}-{app_config.stage}"
+        super().__init__(scope, id, **kwargs)
+
+        discovery_lambda = aws_lambda.Function(
+            self,
+            f"{id}-lambda",
+            runtime=aws_lambda.Runtime.PYTHON_3_12,
+            code=aws_lambda.Code.from_docker_build(
+                path="./infrastructure",
+                file="Dockerfile.api",
+                build_args={
+                    "PYTHON_VERSION": "3.12",
+                    "API_VERSION": app_config.api_version,
+                },
+            ),
+            handler="federated_collection_discovery.main.handler",
+            environment={
+                "FEDERATED_STAC_API_URLS": app_config.stac_api_urls,
+                "FEDERATED_CMR_URLS": app_config.cmr_urls,
+            },
+            memory_size=256,
+            timeout=Duration.seconds(20),
+            log_retention=aws_logs.RetentionDays.ONE_WEEK,
+        )
+
+        if app_config.api_domain_name and app_config.api_certificate_arn:
+            api_certificate = aws_certificatemanager.Certificate.from_certificate_arn(
+                self, "APICertificate", app_config.api_certificate_arn
+            )
+            default_domain_mapping = aws_apigatewayv2.DomainMappingOptions(
+                domain_name=aws_apigatewayv2.DomainName(
+                    self,
+                    "ApiDomainName",
+                    domain_name=app_config.api_domain_name,
+                    certificate=api_certificate,
+                )
+            )
+        else:
+            default_domain_mapping = None
+
+        discovery_api = aws_apigatewayv2.HttpApi(
+            self,
+            f"{id}-endpoint",
+            default_integration=aws_apigatewayv2_integrations.HttpLambdaIntegration(
+                f"{id}-api-integration",
+                handler=discovery_lambda,
+            ),
+            default_domain_mapping=default_domain_mapping,
+        )
+
+        if not discovery_api.url:
+            raise ValueError("there is no url associated with discovery_api!")
+
+        CfnOutput(
+            self,
+            "ApiEndpoint",
+            value=discovery_api.url.strip("/"),
+        )
+
+        # S3 bucket and CloudFront distribution for hosting the client application
+        oai = aws_cloudfront.OriginAccessIdentity(self, f"{id}-oai")
+        client_bucket = aws_s3.Bucket(
```

Comment:
bucket_name would be nice, looks like CloudFormation is making a long resource name